From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Thu, 20 Jan 2011 02:13:33 +0000 (+0100)
Subject: ASoC: ABE DSP: Avoid OFF mode when ABE activity is running
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=ebc7a7aa887d6e027735dbf254f0386922612f61;hp=6ce8e872444996654bf40a1013d6b86c4398d76d

ASoC: ABE DSP: Avoid OFF mode when ABE activity is running

Update ABE driver in order to be sure that OFF mode is
not supportted when ABE is running.

Change-Id: I106d1768c06bd374382471aa4e657690f0041789
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---

Index: Kernel/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- Kernel.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-20 16:19:14.223322565 +0530
+++ Kernel/sound/soc/omap/omap-abe-dsp.c	2011-01-20 16:19:37.383323078 +0530
@@ -43,6 +43,8 @@
 #include <plat/omap_device.h>
 #include <plat/dma.h>
 #include <plat/omap-pm.h>
+#include <plat/powerdomain.h>
+#include "../../../arch/arm/mach-omap2/pm.h"
 
 #include <sound/core.h>
 #include <sound/pcm.h>
@@ -158,6 +160,8 @@
 
 static struct abe_data *abe;
 
+static struct powerdomain *abe_pwrdm;
+
 // TODO: map to the new version of HAL
 static unsigned int abe_dsp_read(struct snd_soc_platform *platform,
 		unsigned int reg)
@@ -1679,6 +1683,9 @@
 	abe_init_engine(platform);
 	abe_add_widgets(platform);
 	abe->platform = platform;
+	abe_pwrdm = pwrdm_lookup("abe_pwrdm");
+	if (!abe_pwrdm)
+		return -ENODEV;
 	return 0;
 }
 
@@ -1785,6 +1792,7 @@
 	pm_runtime_get_sync(&pdev->dev);
 
 	if (!abe->active++) {
+		omap4_set_pwrdm_state(abe_pwrdm, PWRDM_POWER_RET);
 		omap_pm_set_max_dev_wakeup_lat(&pdev->dev, &pdev->dev, 1);
 		aess_restore_context(abe);
 		abe_wakeup();
